src/token/src/canister.rs

```diff
@@ -202,43 +202,47 @@ impl TokenCanister {
         self.state.borrow().stats.owner
     }
 
-    /// Returns an array of transaction records in range [start, start + limit) related to user `who`.
-    /// Unlike `getTransactions` function, the range [start, start + limit) for `getUserTransactions`
-    /// is not the global range of all transactions. The range [start, start + limit) here pertains to
+    /// Returns an array of transaction records in range [start, start + limit] related to user `who`.
+    /// Unlike `getTransactions` function, the range [start, start + limit] for `getUserTransactions`
+    /// is not the global range of all transactions. The range [start, start + limit] here pertains to
     /// the transactions of user who. Implementations are allowed to return less TxRecords than
     /// requested to fend off DoS attacks.
+    ///
+    /// # Arguments
+    /// * `who` - The user to get transactions for.
+    /// * `start` - The index of the first transaction to return.
+    /// * `limit` - The number of transactions to return.
     #[query]
     fn getUserTransactions(&self, who: Principal, start: Nat, limit: Nat) -> Vec<TxRecord> {
-        let mut transactions = vec![];
-
-        let limit_usize = limit.0.to_usize().unwrap_or(usize::MAX);
-        if limit_usize > MAX_TRANSACTION_QUERY_LEN {
-            ic_kit::ic::trap(&format!(
-                "Limit must be less then {}",
-                MAX_TRANSACTION_QUERY_LEN
-            ));
-        }
-
-        for tx in self.state.borrow().ledger.get_range(&start, &limit) {
-            if tx.from == who || tx.to == who || tx.caller == Some(who) {
-                transactions.push(tx.clone());
-            }
-        }
-
-        transactions
-    }
+        // If the start value is larger than usize then return an
+        // empty vec
+        let start = match start.0.to_usize() {
```

Comment:
Why can't this be done with an into?

Reply:
If the start number, which is a `Nat` (and user input) is larger than a `usize` the `to_usize` function will return None. So we either need to default to something or return an empty vec

Reply:
if start is that large we won't have any txs, i think it's ok to .expect("not that big!")

Reply:
or .into()